source/simulator/assembly.cc

```diff
@@ -531,7 +531,7 @@ namespace aspect
                                            cell,
                                            this->introspection,
                                            current_linearization_point,
-                                           need_viscosity);
+                                           /*compute_strain_rate=*/ true);
```

Comment:
Doesn't this negate the point of the flag? Do you even still need `need_viscosity` in this case?

Reply:
We still need the flag, because `need_viscosity` is used in the next line to tell the material model which *outputs* to compute. The wrong assumption here was that if we do not explicitly request the viscosity we also never need the strain rate *input*. This is not true anymore for complex material model (strain rate may be used for other properties than just viscosity).

Reply:
Ah, that's a bummer. I recall that computing the strain rate was expensive, and it's a shame that we now compute it unconditionally because the material model does not have a way to indicate whether or not it actually wants it...